openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIResponseMapper.java

```diff
@@ -490,8 +491,22 @@ private Optional<ApiResponses> getAPIResponses(OperationAdaptor operationAdaptor
             case ERROR_TYPE_DESC:
                 // Return type is given as error or error? in the ballerina it will generate 500 response.
                 apiResponse.description(HTTP_500_DESCRIPTION);
-                mediaType.setSchema(new StringSchema());
-                apiResponse.content(new Content().addMediaType(MediaType.TEXT_PLAIN, mediaType));
+                ObjectSchema errorSchema = new ObjectSchema();
+                // Create ErrorPayload object properties.
+                Map<String, Schema> properties = new HashMap<>();
+                properties.put("timestamp", new StringSchema().description("Timestamp of the error"));
+                properties.put("status", new IntegerSchema().description("Relevant HTTP status code"));
+                properties.put("reason", new StringSchema().description("Reason phrase"));
+                properties.put("message", new StringSchema().description("Error message"));
+                properties.put("path", new StringSchema().description("Request path"));
+                properties.put("method", new StringSchema().description("Method type of the request"));
+                errorSchema.setProperties(properties);
+                if (components.getSchemas() == null) {
+                    components.setSchemas(new HashMap<>());
+                }
+                components.getSchemas().put("ErrorPayload", errorSchema);
+                mediaType.setSchema(new Schema<>().$ref("ErrorPayload"));
```

Comment:
Shouldn't we keep constants for the ErrorPayload name and its object properties?

Reply:
I kept these without moving to constant since these have only one-time usage.